crates/bridge-exec/src/ws_client.rs

```diff
@@ -0,0 +1,158 @@
+//! Wrapper for managing a WebSocket client that allows recycling or restarting
+//! the client without restarting the entire program.
+
+use core::fmt;
+
+use deadpool::managed::{self, Manager, RecycleError, RecycleResult};
+use jsonrpsee::{
+    core::{
+        async_trait,
+        client::{BatchResponse, ClientT},
+        params::BatchRequestBuilder,
+        traits::ToRpcParams,
+        BoxError, ClientError, DeserializeOwned,
+    },
+    ws_client::{WsClient as WebsocketClient, WsClientBuilder},
+};
+
+/// Configuration for the WebSocket client.
+///
+/// settings that are necessary to initialize and configure
+/// the WebSocket client, ie. URL of the WebSocket server.
+#[derive(Clone, Debug)]
+pub struct WsClientConfig {
+    /// The URL of the WebSocket server.
+    pub url: String,
+}
+
+/// Manager for creating and recycling WebSocket clients.
+///
+/// This struct manages the lifecycle of WebSocket clients, including creating
```

Comment:
```suggestion
/// Manages the lifecycle of WebSocket clients, including creating
```

"this struct" is ambiguous. It is already a `pub struct Foo`.